cli/harness/src/main/java/io/helidon/build/cli/harness/CommandParser.java

```diff
@@ -112,6 +120,41 @@ static CommandParser create(String... args) {
         return new CommandParser(argsList, commandName, new Resolver(params, properties), error);
     }
 
+    static String[] preProcessArgs(String[] args) {
+        List<String> result = new ArrayList<>();
+        for (int i = 0; i < args.length; i++) {
+            if ("--args-file".equals(args[i]) && (i < args.length - 1)) {
+                String argsFile = args[i + 1];
+                result.addAll(readArgsFile(argsFile));
+                i++;
+            } else {
+                result.add(args[i]);
+            }
+        }
+        return result.toArray(new String[0]);
+    }
+
+    static List<String> readArgsFile(String argsFile) {
+        List<String> result = new ArrayList<>();
+        try {
+            URL fileURL = new URL(argsFile);
```

Comment:
I don't think, it makes sense to have it be a URL.
You can use `Path.of` (which would support file system related URIs) and feed that into `Files.lines`

Reply:
fixed